xip/xipether.c

```diff
@@ -0,0 +1,449 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <assert.h>
+#include <sys/types.h>
+#include <asm/byteorder.h>
+#include <asm-generic/errno-base.h>
+#include <net/xia_fib.h>
+#include <xia_socket.h>
+#include <sys/socket.h>
+#include <linux/netdevice.h>
+
+#include "xip_common.h"
+#include "utils.h"
+#include "ll_map.h"
+#include "libnetlink.h"
+#include "xiart.h"
+
+#define IFINDEX_STR_SIZE 8
+#define XIA_LLADDR_LEN 12
+#define APPEND_ZEROES 20
+
+static int usage(void)
+{
+	fprintf(stderr,
+"Usage: xip ether { addinterface | delinterface } dev DEV\n"
+"       xip ether { addneigh | delneigh } lladdr LLADDR dev DEV\n"
+"       xip ether show { interfaces | neighs }\n"
+"where  LLADDR := HEXDIGIT{1,2} (':' HEXDIGIT{1,2})*\n"
+"       DEV := STRING NUMBER\n");
+	return -1;
+}
+
+static int form_ether_xid(unsigned oif, unsigned char *lladdr, unsigned tlen,
+			  char *id)
+{
+	char bytes[4];
+	unsigned append = APPEND_ZEROES;
```

Comment:
Let's get rid of `append` as well as the macro `APPEND_ZEROES` defined at the top of the file, and instead do the trick I mentioned below.